### DISCUSSION TITLE
How to use a device pointer in PyCUDA kernel call?

### DISCUSSION THREAD

**Discussion by villekf** [inducer/pycuda#439] [Q&A]
This is a similar question as outlined here: https://github.com/inducer/pycuda/discussions/347#discussion-3895878

I have an Arrayfire array that allows to output the device pointer as `int`. Is it possible to somehow use this device pointer in a kernel call? Or make a GPUArray from it? In PyOpenCL this can be achieved with `cl.MemoryObject.from_int_ptr`, but is there an equivalent function in PyCUDA?

**Comment by pierrepaleo** [comment]
You can create a `GPUArray` from a device pointer, which can be latter passed to kernels.

```python
import numpy as np
import arrayfire as af
if af.get_active_backend() != "cuda":
    af.set_backend("cuda")
import pycuda.gpuarray as garray

af_arr = af.to_array(np.ones(50, dtype="f"))
pycuda_arr = garray.empty(af_arr.shape, np.float32, gpudata=af_arr.device_ptr())
assert pycuda_arr.ptr == af_arr.device_ptr()
pycuda_arr += 1
print(af_arr.to_ndarray())
```

Beware, from my (admittedly very limited) experience with arrayfire, it seems that arrayfire almost-always re-allocates memory (even for in-place operations like `af_arr += np.float32(1)`) ; so the arrayfire pointer is likely to change.
